app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -23,50 +22,45 @@
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 import android.text.Html;
-import android.text.TextUtils;
 import android.util.Base64;
 import android.util.Log;
+import android.view.Gravity;
 import android.view.MenuItem;
 import android.view.View;
+import android.view.ViewParent;
 import android.widget.ArrayAdapter;
-import android.widget.Button;
-import android.widget.CompoundButton;
 import android.widget.DatePicker;
 import android.widget.EditText;
+import android.widget.ImageButton;
 import android.widget.ImageView;
 import android.widget.LinearLayout;
 import android.widget.Spinner;
 import android.widget.TextView;
 import android.widget.Toast;
-import android.widget.ToggleButton;
 
 import com.google.android.gms.common.api.GoogleApiClient;
-import com.google.android.gms.location.LocationServices;
-import com.google.android.gms.location.places.Places;
 import com.google.gson.Gson;
 
 import java.io.ByteArrayOutputStream;
 import java.io.File;
+import java.lang.reflect.Field;
 import java.text.DateFormat;
 import java.util.ArrayList;
 import java.util.Calendar;
-import java.util.Date;
 import java.util.List;
-import java.util.concurrent.ExecutionException;
-
-import layout.homeFragmment;
-import layout.mapFragment;
-import layout.profileFragment;
 
 public class EditMoodActivity extends AppCompatActivity {
 
     public static final int MAX_PHOTO_SIZE = 65536;
+    private static int NUM_EMOTIONS = FeelTripApplication.getNumEmotions();
     private EditText inputMoodDescription;
+    private LinearLayout emojiList;
+    private ImageButton emojiButton;
```

Comment:
Initialize the dynamic EZ loading of the FeelTripApplication List containing all emotions, ordered as we see fit.